src/main/Main.ml

```diff
@@ -96,20 +96,24 @@ let position_for_experimental_flag argv =
 
 (* TODO[Issue #131]: Add some expectation tests for such functions. *)
 let with_experimental_flag argv =
-  let len = position_for_experimental_flag argv in
-  Array.concat [
-    Array.sub argv 0 len;
-    [| "--experimental" |];
-    Array.sub argv len (Array.length argv - len);
-  ]
+  (* Only add the flag if it's not already present *)
+  if Array.mem "--experimental" argv then
+    argv
+  else
+    let len = position_for_experimental_flag argv in
+    Array.concat [
+      Array.sub argv 0 len;
+      [| "--experimental" |];
+      Array.sub argv len (Array.length argv - len);
+    ]
 
 (* let _ = assert (with_experimental_flag [| "opengrep"; "scan"; "--help" |]
                    = [| "opengrep"; "scan"; "--experimental"; "--help" |])
    let _ = assert (with_experimental_flag [| "opengrep"; "-c"; "rules"; "libs" |]
                    = [| "opengrep"; "--experimental"; "-c"; "rules"; "libs" |]) *)
 
 let flags_that_require_experimental : string list =
-  [ "--output-enclosing-context" ]
+  [ "--output-enclosing-context"; "--opengrep-ignore-pattern"; "--incremental-output" ]
```

Comment:
@tom-paz you previously implemented this flag (`--opengrep-ignore-pattern`) on the python side too.

We cannot just require `--experimental` when the python code to implement this has been added, and in fact does not seem to work.

- If we were to only support this with `--experimental`, all the python code would have to be reverted. (Now is too late for that.)
- But since we do have the feature in the python part, we should not require `--experimental` but rather we should also fix the python part too.